recursion.rb

```diff
@@ -1,17 +1,63 @@
+# RC: n! =  n * (n-1)!
+# BC: 1! = 1
+# BC: 0! = 1
 
 def fact(n)
+  if n == 0 || n == 1
+    return 1
+  end
+ return n * fact(n - 1)
 end
 
+# 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89
+# example: fib(4) = 3
+# BC: fib(1) = 1
+# BC: fib(2) = 1
+# exercises: fib(3) = fib(2) + fib(1)
+#            fib(4) = fib(3) + fib(2)
+
+# fib(n) = fib(n-1) + fib(n-2)
 def fib(n)
+  if n == 1 || n == 2
+    return 1
+  end
+  return fib(n-1) + fib(n-2)
 end
 
-def pal(s)
+def binary(n)
+  #First param = number of digits we're expecting. Second param = binary being constructed. Third param = entire sequence of binary numbers
+  binary_helper(n,"","")
 end
 
-def binary(n)
+def binary_helper(digits, binary, result)
+  if digits == 0
+    result += binary + " "
+    return result
+  end
+  result = binary_helper(digits-1, binary+"0", result)
+  result = binary_helper(digits-1, binary+"1", result)
+  return result
 end
 
+#BC - Any single character string is a palindrome.
+#HOMEWORK
+def pal(s)
+  array = s.split(//)
+  if array.length == 1 || array.length == 0
+    return true
+  elsif array[0] != array[-1]
+    return false
+  elsif array[0] == array[-1]
+   array.shift
+   array.pop
+   s = array.join
+   pal(s)
+  end
+end
```

Comment:
I like the readability of this solution -- though it does use auxillary storage (an array) which wouldn't have been needed if you applied the same logic to string slicing instead of converting to an array and then indexing.